python/pyspark/ml/pipeline.py

```diff
@@ -89,9 +87,9 @@ def setParams(self, stages=None):
         setParams(self, stages=None)
         Sets params for Pipeline.
         """
-        if stages is None:
-            stages = []
         kwargs = self.setParams._input_kwargs
+        if stages is None:
+            kwargs['stages'] = []
```

Comment:
Do we need this? Now that we've got `getOrDefault` inside of `getStages` it shouldn't be necessary provided we use the get function.